### DISCUSSION TITLE
Env-Variable - Renaming SMTP_EXPLICIT_TLS

### DISCUSSION THREAD

**Discussion by sando38** [dani-garcia/vaultwarden#2226] [Q&A]
Hello there,

reporting this, because I missinterpreted it. According to RFC8314 (https://datatracker.ietf.org/doc/html/rfc8314#section-3.3) implicit TLS is used for port 465.
However, to force tls with environment variables in the smtp settings, I have to set:

SMTP_EXPLICIT_TLS=true -> which would actually imply STARTTLS.

So, either I suggest renaming the environment variable into:

SMTP_IMPLICIT_TLS or 
SMTP_FORCE_TLS

to avoid confusion.

NB: Thanks for your great service with providing vaultwarden. It is a great tool.

Wishing a nice day!

**Comment by BlackDex** [comment]
This will be a wont fix, since this would lead to a lot of invalid configurations. A lot of people have this configured using that variable and we would not like to break those. Also, adding a second variable and keeping the current one will probably lead to confusion or also some weird issues.

There is already a mentioning of the wrongly named variable here: https://github.com/dani-garcia/vaultwarden/blob/main/.env.template#L335

Also see #851 (which this ticket is a duplicate of)

**Reply by sando38** [reply]
Alrighty, did not scroll in depth into the provided env file template, only looked into the wiki. Maybe it is worth to mention in the wiki as well.

**Comment by BlackDex** [comment]
Please update it! Be our guest

**Reply by sando38** [reply]
done!

**Comment by salixh5** [comment]
The config file states `SMTP_SSL Needs to be set to true for this option to work.` when talking about the improperly labeled option for implicit TLS. However, the wiki page https://github.com/dani-garcia/vaultwarden/wiki/SMTP-Configuration says that "for mail servers that use port 465" to use `SMTP_SSL=false`.

Could this be unified so it is clear what settings to use here? I'm trying to speak to my SMTP relay server on port 465 via implicit TLS. Do I need to set:
```
SMTP_PORT=465
SMTP_SSL=false
SMTP_EXPLICIT_TLS=true
```

Or should I set:
```
SMTP_PORT=465
SMTP_SSL=true
SMTP_EXPLICIT_TLS=true
```

**Reply by BlackDex** [reply]
As stated in: https://github.com/dani-garcia/vaultwarden/wiki/SMTP-Configuration
It should be the first one.

And if i read the comments placed after those specific config variables it also mentions which ones are needed for 465 or 587.
So i think it should be very clear, besides the fact that they are incorrectly labeled  unfortunately.

**Reply by salixh5** [reply]
Hey BlackDex,

then this comment [# SMTP_EXPLICIT_TLS=true # (Implicit) - N.B. This variable configures Implicit TLS. It's currently mislabelled (see bug #851) - SMTP_SSL Needs to be set to true for this option to work. Usually port 465 is used here. ](https://github.com/dani-garcia/vaultwarden/blob/08f0de7b46d36a4e974d8e7b25a9786168cd38a5/.env.template#L336) is wrong and should read "Needs to be set to false for this option to work"

**Reply by BlackDex** [reply]
Ah... Well, actually the term `needs` needs to be removed. Because it doesn't need to be anymore, it may though, but it doesn't conflict or change the outcome :).

**Comment by pepa65** [comment]
What happens if this is set?
SMTP_SSL=true
SMTP_EXPLICIT_TLS=true

Is there any valid use case for it, or is it always wrong??

**Reply by BlackDex** [reply]
In the past you needed to set `SMTP_SSL` to true, else the other option would not do anything. I changed that a long while back that it doesn't matter which one you set. Though, `SMTP_EXPLICIT_TLS` will overrule and make the connection Full SSL only, no StartTLS.

Maybe we need to rename these variables when we release the async version, since that will be a major change anyway. And generate some alerts if the old ones are still used.
